cli/lsp/language_server.rs

```diff
@@ -778,6 +819,210 @@ impl Inner {
     }
   }
 
+  async fn code_lens(
+    &mut self,
+    params: CodeLensParams,
+  ) -> LspResult<Option<Vec<CodeLens>>> {
+    if !self.enabled() || !self.config.settings.enabled_code_lens() {
+      return Ok(None);
+    }
+
+    let mark = self.performance.mark("code_lens");
+    let specifier = utils::normalize_url(params.text_document.uri);
+    let line_index = self.get_line_index_sync(&specifier).unwrap();
+    let navigation_tree =
+      self.get_navigation_tree(&specifier).await.map_err(|err| {
+        error!("Failed to retrieve nav tree: {:#?}", err);
+        LspError::invalid_request()
+      })?;
+
+    let cl = Rc::new(RefCell::new(Vec::new()));
```

Comment:
The use of Rc and RefCell here seem a bit questionable. But I assume you tried this with `let mut cl = Vec::new()` already and the types didn't allow it.

Reply:
Yes of course.  The closure below gives the compiler the heebie-jeebies when dealing with the mutable vector, so it isn't questionable, it is required when dealing with stuff in closures that you aren't moving in there.